File: src/main/java/net/alephdev/calendar/models/Idea.java
package net.alephdev.calendar.models;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "Idea")
@SuppressFBWarnings({"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
public class Idea {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String description;

  @ManyToOne
  @JoinColumn(name = "author_login", foreignKey = @ForeignKey(name = "fk_idea_author_login"))
  private User authorLogin;

  @Enumerated(EnumType.STRING)
  @Column(name = "status_enum_id", nullable = false)
  private Status statusEnumId;

  @ManyToOne
  @JoinColumn(name = "task_id", foreignKey = @ForeignKey(name = "fk_idea_task_id"))
  private Task task;

  public enum Status {
    PENDING,
    REJECTED,
    APPROVED;
  }
}

File: angular/src/app/models/idea.ts
import {User} from "./user";
import {Task} from "./task";

/**
 * Модель идеи (предложения/фичи) в системе.
 *
 * Идея может быть связана с задачей и иметь статус модерации.
 *
 * @property id - Уникальный идентификатор идеи
 * @property description - Описание идеи
 * @property authorLogin - Автор идеи
 * @property statusEnumId - Статус идеи
 * @property task - Связанная задача (если идея принята в работу)
 */
export interface Idea {
  id: number;
  description: string;
  authorLogin: User;
  statusEnumId: IdeaStatus;
  task: Task | null;
}

/**
 * Статусы идеи.
 *
 * @enum {string}
 * @property PENDING - Ожидает решения
 * @property REJECTED - Отклонена
 * @property APPROVED - Принята
 */
export enum IdeaStatus {
  PENDING = 'PENDING',
  REJECTED = 'REJECTED',
  APPROVED = 'APPROVED'
}
